src/cpp/rtps/builtin/discovery/endpoint/EDPServer.cpp

```diff
@@ -283,9 +283,9 @@ bool EDPServer::removeLocalReader(RTPSReader* R)
     {
         InstanceHandle_t iH;
         iH = (R->getGuid());
-        CacheChange_t* change = writer->first->new_change([]() -> uint32_t 
+        CacheChange_t* change = writer->first->new_change([this]() -> uint32_t
             {
-                return DISCOVERY_SUBSCRIPTION_DATA_MAX_SIZE;
+                return mp_PDP->builtin_attributes().readerPayloadSize;
```

Comment:
This would be `writerPayloadSize`, as we are adding the change to the built-in writer.